### PR TITLE
feat: add support for versioned secrets in shared access key auth

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add support for versioned secrets in the shared access key authentication.

### CODE REVIEW

**Comment by stijnmoreels** [comment]
After this, we could provide a preview package so that Gillian already can take a look.

File: docs/preview/03-Features/security/auth/shared-access-key.md [hunk 1]
```diff
@@ -79,6 +67,8 @@ builder.Services.AddControllers(options =>
 For this setup to work, an Arcus secret store is required as the provided secret name (in this case `"shared-access-key-name"`) will be looked up.
 See [our official documentation](https://security.arcus-azure.net/features/secret-store/) for more information about setting this up.
 
+> 💡 This authentication mechanism supports multiple secret versions so that many different versions of a single configured secret are checked during authentication. See [our official documentation]() for more information on how to configured your secret as 'versioned' in the secret store.
```
Comment (stijnmoreels) [inline docs/preview/03-Features/security/auth/shared-access-key.md:70]: Build failed bc we can't fill this out yet, as the security docs are not yet released: https://github.com/arcus-azure/arcus.security/pull/363

**Comment by fgheysels** [comment]
To be honest, I'm currently really wondering what kind of problem this 'versioned secret' thing solves.
When you update a certain key for instance, the consumer of the resource for which you need that key, will need to use the most recent secret.   I don't see why you would -in a normal situation- need to fall back to an older version of a secret ?

**Comment by stijnmoreels** [comment]
> To be honest, I'm currently really wondering what kind of problem this 'versioned secret' thing solves. When you update a certain key for instance, the consumer of the resource for which you need that key, will need to use the most recent secret. I don't see why you would -in a normal situation- need to fall back to an older version of a secret ?

This mostly came from a feature request from @gverstraete, https://github.com/arcus-azure/arcus.webapi/issues/358

**Review by fgheysels — APPROVED** [review]
I approve this, although I don't see much use for this feature to be honest.
When a new version of a secret is added in KeyVault, you typically want to use that new version of a secret, and not the previous one.
When a new version is added, it means for instance the key that must be used to access a certain resource has been updated.  So I don't see no reason to be able to fall back on the previous version of the scret :?

**Comment by stijnmoreels** [comment]
> I approve this, although I don't see much use for this feature to be honest. When a new version of a secret is added in KeyVault, you typically want to use that new version of a secret, and not the previous one. When a new version is added, it means for instance the key that must be used to access a certain resource has been updated. So I don't see no reason to be able to fall back on the previous version of the scret :?

Think this is the main request that is being supported here: "We are working on a key rotation system, which means that when a new secret version is deployed, the new one & the old one should be still active & be considered as "valid".."

**Pull request merged**